packages/openapi-generator/src/parser/operation.ts

```diff
@@ -64,102 +59,86 @@ export function getRelevantParameters(
-// 2. `/path?{param}`
-function isValidPathPattern(
+/**
+ * @internal
+ */
+export function parsePathPattern(
   pathPattern: string,
-  placeholders: string[]
-): boolean {
-  return (
-    pathPattern.includes('?') ||
-    placeholders.some(placeholder => !isValidPlaceholder(placeholder))
-  );
-}
-
-function sortPathParameters(
-  pathParameters: OpenAPIV3.ParameterObject[],
-  pathPattern: string
-): OpenAPIV3.ParameterObject[] {
-  const pathParts = pathPattern.split('/');
-  const placeholders = pathParts.filter(part => isPlaceholder(part));
+  pathParameters: OpenApiParameter[]
+): string {
+  // Get the innermost curly bracket pairs with non-empty and legal content as placeholders.
+  const placeholders = pathPattern.match(/{[^/?#{}]+}/g);
+  // Get the non-parameter strings as static parts
+  const staticParts = pathPattern.split(/{[^/?#{}]+}/);
 
-  if (isValidPathPattern(pathPattern, placeholders)) {
-    throw new Error(
-      `Path pattern '${pathPattern}' is invalid or not supported.`
-    );
+  if (!placeholders) {
+    // No placeholder or path parameter found.
+    if (pathParameters.length > 0) {
+      throw new Error(
+        `Could not find placeholder for path parameter '${pathParameters
+          .map(pathParameter => pathParameter.originalName)
+          .join("', '")}'.`
+      );
+    }
+    return pathPattern;
   }
 
-  return placeholders.map(placeholder => {
+  const sortedPlaceholders = placeholders.map(placeholder => {
     const strippedPlaceholder = placeholder.slice(1, -1);
     const pathParameter = pathParameters.find(
-      param => param.name === strippedPlaceholder
+      param => param.originalName === strippedPlaceholder
     );
     if (!pathParameter) {
       throw new Error(
-        `Path parameter '${strippedPlaceholder}' provided in path is missing in path parameters.`
+        `Could not find path parameter for placeholder '{${strippedPlaceholder}}'.`
       );
     }
-
-    return pathParameter;
+    return `{${pathParameter.name}}`;
   });
-}
-/**
- * @internal
- */
-export function parsePathPattern(
-  pathPattern: string,
-  pathParameters: OpenApiParameter[]
-): string {
-  const pathParts = pathPattern.split('/');
-  const parameterNames = pathParameters.map(param => param.name);
 
-  return pathParts
-    .map(part => {
-      if (isPlaceholder(part)) {
-        if (!parameterNames.length) {
-          throw new Error(
-            `Could not find parameter for placeholder '${part}'.`
-          );
-        }
-        return `{${parameterNames.shift()}}`;
-      }
-      return part;
-    })
-    .join('/');
+  // Check if all path parameters match placeholders
+  for (const pathParameter of pathParameters) {
+    if (
+      !placeholders.find(
+        placeholder => placeholder.slice(1, -1) === pathParameter.originalName
+      )
+    ) {
+      throw new Error(
+        `Could not find placeholder for path parameter '${pathParameter.originalName}'.`
+      );
+    }
+  }
```

Comment:
I applied your suggestion. Thanks!

Reply:
There were some complications though. I think we should use `missingPlaceholders` and `originalParameterNames.filter()` since we want to say `could not find placeholder...`. And `includes()` should be negated.

Still, I like the idea because it checks for all missing placeholders. 👍